lib/datadog/appsec/actions_handler.rb

```diff
@@ -19,7 +22,33 @@ def interrupt_execution(action_params)
         throw(Datadog::AppSec::Ext::INTERRUPT, action_params)
       end
 
-      def generate_stack(_action_params); end
+      def generate_stack(action_params)
+        return unless Datadog.configuration.appsec.stack_trace.enabled
+
+        context = AppSec.active_context
+        if context.nil? || context.trace.nil? && context.span.nil?
+          Datadog.logger.debug { 'Cannot find trace or service entry span to add stack trace' }
+          return
+        end
+
+        config = Datadog.configuration.appsec.stack_trace
+
+        # Check that the sum of stack_trace count in trace and entry_span does not exceed configuration
+        span_stack = ActionsHandler::StackTraceInMetastruct.create(context.span&.metastruct)
+        trace_stack = ActionsHandler::StackTraceInMetastruct.create(context.trace&.metastruct)
+        return if config.max_collect != 0 && span_stack.count + trace_stack.count >= config.max_collect
+
+        # Generate stacktrace
+        utf8_stack_id = action_params['stack_id'].encode('UTF-8') if action_params['stack_id']
```

Comment:
This variable will be `nil` if no `params[stack_id]` is given, right? I think the whole method will not work without stack_id, hence we always have it or of not - we should do nothing and report a telemetry or generate warning or so.

Because of that that line must be adjusted to

```suggestion
        stack_id = action_params['stack_id'].encode('UTF-8')
```
_P.S it doesn't make sense to mention UTF-8 in the variable name because it's not important_